Release notes for 2.0.0 release (#658)
kyma-project/api-gateway: docs/release-notes/2.0.0.md

> After much anticipation, we proudly announce the first production release of the API Gateway module as the 2.0.0 relase of API Gateway. The release includes the APIRule controller as before, but also adds support for the new API Gateway module and its corresponding custom resource. The module will now be added to the collection of other modules that have already been separated from the classic Kyma as part of the modularization process. The API Gateway module offers similar functionality to the API Gateway reconciler. To better understand its capabilities, read the documentation. And now, the time has come for you to install the Istio module and share your feedback with us!

Reviewer: ```suggestion
After much anticipation, we are excited to announce the release of version 2.0.0 of API Gateway. This is the first production release of the API Gateway module, which offers similar functionality to the API Gateway reconciler. In addition to APIRule Controller that was previously introduced, this release also includes support for the new module and its custom resource.
The API Gateway module will be added to [the collection of other modules](https://kyma-project.io/#/06-modules/README) that have already been separated from the classic Kyma as part of the modularization process. To better understand its capabilities, read the [documentation](https://github.com/kyma-project/api-gateway/tree/main/docs/user#readme). 
And now, the time has come for you to install the API Gateway module and share your feedback with us!
```

## New features

> - Introduces API Gateway CRD

Reviewer: ```suggestion
- Introduce API Gateway CRD
```

> - Module installs Ory Oathkeeper v0.38.25 beta.1

Reviewer: ```suggestion
- The module installs Ory Oathkeeper v0.38.25 beta.1
```

> - Module deploys default Kyma Gateway if enabled in API Gateway CR

Reviewer: ```suggestion
- If enabled in the API Gateway CR, the module deploys the default Kyma Gateway
```

> - Support for API Gateway CR blocking deletion strategy when APIRules, ORY Oathkeeper Rules or Istio VirtualServices exist on the cluster

Reviewer: ```suggestion
- Add support for API Gateway CR blocking deletion strategy when APIRules, Ory Oathkeeper Access Rules, or Istio Virtual Services exist on the cluster
```
